crates/next-core/src/segment_config.rs

```diff
@@ -0,0 +1,1264 @@
+            IssueSeverity::Error,
+        )
+        .await?;
+    }
+
+    Ok(config.cell())
+}
+
+async fn invalid_config(
+    source: ResolvedVc<Box<dyn Source>>,
+    key: &str,
+    span: Span,
+    error: RcStr,
+    value: Option<&JsValue>,
+    severity: IssueSeverity,
+) -> Result<()> {
+    let detail = if let Some(value) = value {
+        let (explainer, hints) = value.explain(2, 0);
+        Some(*StyledString::Text(format!("Got {explainer}.{hints}").into()).resolved_cell())
+    } else {
+        None
+    };
+
+    NextSegmentConfigParsingIssue::new(
+        source.ident(),
+        key.into(),
+        error,
+        detail,
+        IssueSource::from_swc_offsets(source, span.lo.to_u32(), span.hi.to_u32()),
+        severity,
+    )
+    .to_resolved()
+    .await?
+    .emit();
+    Ok(())
+}
+
+async fn parse_config_value(
+    source: ResolvedVc<Box<dyn Source>>,
+    is_app_router: bool,
+    config: &mut NextSegmentConfig,
+    eval_context: &EvalContext,
+    key: &str,
+    init: Option<Cow<'_, Expr>>,
+    span: Span,
+) -> Result<()> {
+    let get_value = || {
+        let init = init.as_deref();
+        // Unwrap `export const config = { .. } satisfies MiddlewareConfig`, usually this is already
+        // transpiled away, but we are looking at the original source here.
+        let init = if let Some(Expr::TsSatisfies(TsSatisfiesExpr { expr, .. })) = init {
+            Some(&**expr)
+        } else {
+            init
+        };
+        init.map(|init| eval_context.eval(init)).map(|v| {
+            // Special case, as we don't call `link` here: assume that `undefined` is a free
+            // variable.
+            if let JsValue::FreeVar(name) = &v
+                && name == "undefined"
+            {
+                JsValue::Constant(ConstantValue::Undefined)
+            } else {
+                v
+            }
+        })
+    };
+
+    match key {
+        "config" => {
+            let Some(value) = get_value() else {
+                return invalid_config(
+                    source,
+                    "config",
+                    span,
+                    rcstr!("It mustn't be reexported."),
+                    None,
+                    IssueSeverity::Error,
+                )
+                .await;
+            };
+
+            if is_app_router {
+                return invalid_config(
+                    source,
+                    "config",
+                    span,
+                    rcstr!(
+                        "Page config in `config` is deprecated and ignored, use individual \
+                         exports instead."
+                    ),
+                    Some(&value),
+                    IssueSeverity::Warning,
+                )
+                .await;
+            }
+
+            let JsValue::Object { parts, .. } = &value else {
+                return invalid_config(
+                    source,
+                    "config",
+                    span,
```

Comment:
nit: use the span for the init expression

Reply:
Once we enter `JSValue` land, we don't have spans anymore, unfortunately. If I understand you correctly

Reply:
well for this one i think you could `init.map_or_default(|init|init.span(), span())` and at least point at the Left hand side, but not a big deal

Reply:
I intententially changed this further up in the file to not use the init span here:
```
                                    // The config object can span hundreds of lines. Don't
                                    // highlight the whole thing
```

It's just way too much noise if you have 100 lines of config. There is no additional information here by printing 100 lines of codeframe.

<img width="1564" height="728" alt="Bildschirmfoto 2025-09-03 um 23 15 46" src="https://github.com/user-attachments/assets/507270ca-09a2-4492-bd8b-6c4cc600326c" />